### PR TITLE
Ignore less variables (#4200)

### LINKED ISSUE #4200: Fix false positives for Less variables in length-zero-no-unit
> Clearly describe the bug

Automatic fix (` --fix`) for `length-zero-no-unit` does nothing when working with Less variables.

> Which rule, if any, is the bug related to?

`length-zero-no-unit`

> What CSS is needed to reproduce the bug?

```less
@variable: 0px;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "length-zero-no-unit": true
  }
}
```
> Which version of stylelint are you using?

`10.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to Less variables.

> What did you expect to happen?

Input file to be fixed:
```less
@variable: 0;
```

> What actually happened (e.g. what warnings or errors did you get)?

Input file did not change.

### PR DESCRIPTION
**Pull request by fanich37**
> Which issue, if any, is this issue related to?

It should close #4200 .

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/length-zero-no-unit/index.js [hunk 1]
```diff
@@ -65,6 +65,10 @@ const rule = function(actual, secondary, context) {
 		});
 
 		root.walkAtRules((atRule) => {
+			if (atRule.variable && !hasBlock(atRule)) {
```
Comment (hudochenkov) [inline lib/rules/length-zero-no-unit/index.js]: Could you add comment before this condition to specify that this condition is for Less variables.

Comment (fanich37) [reply lib/rules/length-zero-no-unit/index.js]: Done!

File: lib/rules/length-zero-no-unit/index.js [hunk 2]
```diff
@@ -65,6 +65,11 @@ const rule = function(actual, secondary, context) {
 		});
 
 		root.walkAtRules((atRule) => {
+			// Ignore Less variables
+			if (atRule.variable && !hasBlock(atRule)) {
+				return;
+			}
```
Comment (evilebottnawi) [inline lib/rules/length-zero-no-unit/index.js]: Maybe we should move this to utils?

Comment (fanich37) [reply lib/rules/length-zero-no-unit/index.js]: Sounds reasonable.
Added util function and some tests.

**Comment by hudochenkov** [comment]
> -   Fixed: `length-zero-no-unit` false positives for Less variables ([#4405](https://github.com/stylelint/stylelint/pull/4405)).

**Pull request merged**